scripts.js

```diff
@@ -9,15 +17,121 @@ container1.appendChild(tile);
 // Your implementation goes here
 
 function createTileObject(name, top, left) {
-  const newTile = document.createElement('div');
-  newTile.classList.add('tile');
-  newTile.innerText = name;
-  newTile.style.top = top + 'px';
-  newTile.style.left = left + 'px';
+    const newTile = document.createElement('div');
+    newTile.classList.add('tile');
+    newTile.innerText = name;
+    newTile.style.top = top + 'px';
+    newTile.style.left = left + 'px';
+    newTile.id = idCounter;
+    newTile.addEventListener('click', function () {
+        if (this.parentElement === container1) {
+            moveTile(this.id, this.parentElement, container2);
+        } else {
+            moveTile(this.id, this.parentElement, container1);
+        }
+    });
+    newTile.draggable = 'true';
+    newTile.addEventListener('dragstart', function (ev) {
+        ev.dataTransfer.setData('div', ev.target.id);
+    });
 
-  return newTile;
+    idCounter = idCounter + 1;
+
+    return newTile;
+}
+
+function addTile() {
+    const input = document.getElementById('tile-name').value;
+
+    if (input !== '') {
+        const randomTop = Math.floor(
+            Math.random() * (containerHeight - tileHeight)
```

Comment:
although this is a simple calculation, it is nice to extract in a function (generating a random number in a range), as if you find a bug in the implementation, you will fix it in one place.